### PR TITLE
Feat/multiclass metrics

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Describe your changes

Setting up model types to enable multiclass classification metric routing.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

**Comment by IanAtCredo** [comment]
To do:
* add multiclass classification metrics to list
* Route a multiclass classification model (or dataset?) to the appropriate metrics
    * find_metrics has a `metric_category` attribute that we can probably use. We'd need a helper function to go from a list of artifacts -> metric category
* better messaging if metrics fail

Questions:
* How do we want to deal with models that we can't identify? For instance, if the model is non-sklearn we won't know what kind of classification it is.  Options:
     * We could update the type based on what we see in the dataset (there is a `type_of_target` function which could inform things). That may be a nice thing to do!
     * Classification type routes to binary classification and we have a good warning?

File: credoai/modules/constants_metrics.py [hunk 1]
```diff
@@ -19,10 +19,13 @@
 
 THRESHOLD_METRIC_CATEGORIES = ["BINARY_CLASSIFICATION_THRESHOLD"]
 
-MODEL_METRIC_CATEGORIES = [
+MODEL_TYPE_METRIC_CATEGORIES = [
```
Comment (IanAtCredo) [inline credoai/modules/constants_metrics.py:22]: If you think there's gonna be a close correspondence between metric types and model categories, why not just reference the model category constants defined here?`credoai/artifacts/model/constants_model.py`

image

**Comment by fabrizio-credo** [comment]
- Completed update of MULTICLASS_CLASSIFICATION_FUNCTIONS
- Created our own version for TPR, TNR

**Comment by fabrizio-credo** [comment]
Merged into  feat/test_expansion 166d214841774c099de816fc4b963becd7cd1e52, we can close this.
